common/app/model/content.scala

```diff
@@ -82,6 +83,7 @@ final case class Content(
   lazy val isHeroic = HeroicTemplateSwitch.isSwitchedOn && tags.isLabourLiverpoolSeries
   lazy val isImmersive = fields.displayHint.contains("immersive") || isImmersiveGallery || tags.isUSMinuteSeries || isHeroic
   lazy val isAdvertisementFeature: Boolean = tags.tags.exists{ tag => tag.id == "tone/advertisement-features" }
+  lazy val campaigns: List[Campaign] = targeting.CampaignAgent.getCampaignsForTags(tags.tags.map(_.id))
```

Comment:
This is happening on every origin request.  Is there some limit enforced on the number and complexity of campaigns existing in the s3 bucket?  as we're basically only using lists everywhere so the time complexity will be about O(nrules*(nlacking+nrequired)*ncontenttags*campaigns)
This is fine if we have enforced limits on all of those things, but if we want any of those to have arbitrary length, we need a better algorithm.